### PR TITLE
Move XPU/Opt related PRs from releases2.0 to develop

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary

Merge back https://github.com/openvinotoolkit/training_extensions/pull/3218 and https://github.com/openvinotoolkit/training_extensions/pull/3094 from releases to develop

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/engine/engine.py [hunk 1]
```diff
@@ -878,6 +882,14 @@ def _build_trainer(self, **kwargs) -> None:
         """Instantiate the trainer based on the model parameters."""
         if self._cache.requires_update(**kwargs) or self._trainer is None:
             self._cache.update(**kwargs)
+            # set up xpu device
+            if self._device.accelerator == DeviceType.xpu:
+                self._cache.update(strategy="xpu_single")
+                # add plugin for Automatic Mixed Precision on XPU
+                if self._cache.args["precision"] == 16:
```
Comment (eunwoosh) [inline src/otx/engine/engine.py]:
```suggestion
                if self._cache.args.get("precision", 32) == 16:
```
Some models don't set precision in configuration, which can make an error.

File: src/otx/algo/strategies/xpu_single.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        """Sets up optimizers."""
+        super().setup_optimizers(trainer)
+        if len(self.optimizers) != 1:  # type: ignore[has-type]
+            msg = "XPU strategy doesn't support multiple optimizers"
+            raise RuntimeError(msg)
+        if trainer.task != "SEMANTIC_SEGMENTATION":
+            model, optimizer = torch.xpu.optimize(trainer.model, optimizer=self.optimizers[0])  # type: ignore[has-type]
+            self.optimizers = [optimizer]
+            self.model = model
+
```
Comment (eunwoosh) [inline src/otx/algo/strategies/xpu_single.py]:
```suggestion
        if len(self.optimizers) > 1:  # type: ignore[has-type]
            msg = "XPU strategy doesn't support multiple optimizers"
            raise RuntimeError(msg)
        if trainer.task != "SEMANTIC_SEGMENTATION":
            if len(self.optimizers) == 1:  # type: ignore[has-type]
                model, optimizer = torch.xpu.optimize(trainer.model, optimizer=self.optimizers[0])  # type: ignore[has-type]
                self.optimizers = [optimizer]
                self.model = model
            else:  # for inference
                trainer.model.eval()
                self.model = torch.xpu.optimize(trainer.model)
```
Could you change it like this as #3293?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left some comments. Please take a look.

File: src/otx/algo/strategies/xpu_single.py (hunk 2 above)
Comment (kprokofi) [reply src/otx/algo/strategies/xpu_single.py]: Maybe we can add your PR to develop instead of it?

Comment (kprokofi) [reply src/otx/algo/strategies/xpu_single.py]: I will do that

Comment (kprokofi) [reply src/otx/algo/strategies/xpu_single.py]: Done https://github.com/openvinotoolkit/training_extensions/pull/3295/commits/023cc8f51a90a4c20329bdc67a7c3a3310db5936

File: src/otx/engine/engine.py (hunk 1 above)
Comment (kprokofi) [reply src/otx/engine/engine.py]: Done https://github.com/openvinotoolkit/training_extensions/pull/3295/commits/023cc8f51a90a4c20329bdc67a7c3a3310db5936

**Review by eunwoosh — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Comment by kprokofi** [comment]
seems like tox is broken. pytest works well

File: src/otx/core/types/__init__.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 from typing_extensions import TypeAlias
 
+from otx.core.types.export import TaskLevelExportParameters
```
Comment (harimkang) [inline src/otx/core/types/__init__.py]: I'm looking into why test doesn't work in this PR. 
With the addition of this, it looks like torch is being imported somewhere before `otx install`.
Could you please check it out?

Comment (kprokofi) [reply src/otx/core/types/__init__.py]: Thanks!

**Review by eunwoosh — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**